android/src/androidTest/java/io/rownd/android/AuthInstrumentedTest.kt

```diff
@@ -261,4 +262,96 @@ class AuthInstrumentedTest {
+                }
+            """.trimIndent()
+                    )
+            )
+
+        val initialAccessToken = jwtGenerator.generateTestJwt(
+            expires = Date.from(Instant.now().plusSeconds(55))
+        )
+
+        val storeField = rownd.stateRepo.javaClass.getDeclaredField("store")
+        storeField.isAccessible = true
+        (storeField.get(rownd.stateRepo) as Store<GlobalState, StateAction>).dispatch(
+            StateAction.SetAuth(AuthState(
+                accessToken = initialAccessToken,
+                refreshToken = jwtGenerator.generateTestJwt()
+            ))
+        )
+
+        val resp = rownd.authRepo.getAccessToken()
+
+        assertTrue(resp is String)
+        assertNotSame(initialAccessToken, resp)
+    }
+
+    @Test
+    fun detect_not_expired_access_token() = runTest {
+        server.enqueue(MockResponse()
+            .setHeader("Content-Type", "application/json; charset=utf-8")
+            .setResponseCode(200)
+            .setBody("""
+                {
+                    "access_token": "${jwtGenerator.generateTestJwt()}",
+                    "refresh_token": "${jwtGenerator.generateTestJwt()}"
+                }
+            """.trimIndent())
+        )
+
+        val initialAccessToken = jwtGenerator.generateTestJwt(
+            expires = Date.from(Instant.now().plusSeconds(65))
+        )
+
+        val storeField = rownd.stateRepo.javaClass.getDeclaredField("store")
+        storeField.isAccessible = true
+        (storeField.get(rownd.stateRepo) as Store<GlobalState, StateAction>).dispatch(
+            StateAction.SetAuth(AuthState(
+                accessToken = initialAccessToken,
+                refreshToken = jwtGenerator.generateTestJwt()
+            ))
+        )
+
+        val resp = rownd.authRepo.getAccessToken()
+
+        assertTrue(resp is String)
+        assertEquals(initialAccessToken, resp)
+    }
+
+    @Test
+    fun detect_is_access_token_expired() = runTest {
+        val isExpiredAccessTokenNew = rownd.authRepo.isJwtExpiredWithMargin(JWT(jwtGenerator.generateTestJwt(
+            expires = Date.from(Instant.now().plusSeconds(60*60))
+        )))
+
+        val isExpiredAccessToken65secs = rownd.authRepo.isJwtExpiredWithMargin(JWT(jwtGenerator.generateTestJwt(
+            expires = Date.from(Instant.now().plusSeconds(65))
+        )))
+
+        val isExpiredAccessTokenOld = rownd.authRepo.isJwtExpiredWithMargin(JWT(jwtGenerator.generateTestJwt(
+            expires = Date.from(Instant.now().minusSeconds(60*60))
+        )))
+
+        val isExpiredAccessToken55secs = rownd.authRepo.isJwtExpiredWithMargin(JWT(jwtGenerator.generateTestJwt(
+            expires = Date.from(Instant.now().plusSeconds(55))
+        )))
+
+        assertFalse(isExpiredAccessTokenNew)
+        assertFalse(isExpiredAccessToken65secs)
+
+        assertTrue(isExpiredAccessTokenOld)
+        assertTrue(isExpiredAccessToken55secs)
+    }
```

Comment:
I think we could use a few simple tests where we do the following:
1. Create an access token (expired, nearly expired, newly created)
2. Set the access token on the store
3. Assert that `state.auth.isAccessTokenValid` shows the correct value based on the age of the token. This is the key check for knowing whether a refresh will occur.

Reply:
Would you want me to update the logic for `state.auth.isAccessTokenValid` so that it includes the 60 sec margin? A customer would use this value to trigger a refresh right?